starknet_devnet/blocks.py

```diff
@@ -311,13 +306,20 @@ def get_state(self, block_hash: int) -> StarknetState:
         """Return state at block with `number`"""
         return self.__state_archive.get(block_hash)
 
+    @staticmethod
+    def get_numeric_hash(block_hash: int):
+        """Get numeric hash."""
+        return _parse_block_hash(block_hash)
+
     async def abort_block_by_hash(self, block_hash: str) -> str:
         """
         Abort block by given block hash.
         """
         numeric_hash = _parse_block_hash(block_hash)
 
-        if numeric_hash in self.__hash2block:
+        if numeric_hash in self.__hash2block and numeric_hash in list(
+            self.__num2hash.values()
+        ):
```

Comment:
```suggestion
        if numeric_hash in self.__hash2block and numeric_hash in self.__num2hash.values():
```

Reply:
Wait actually, this will iterate through all `__num2hash` values for each block being aborted? I think a more efficient way is to have a dict of aborted blocks (e.g. `__hash2block_aborted`) and just check if the requested hash is not in there.

Reply:
Check is one thing but the second thing is that we want to remove it later ```del self.__num2hash[block_dict["block_number"]]```.